src/main/java/frc/robot/commandGroups/CommandGroups.java

```diff
@@ -0,0 +1,93 @@
+import frc.robot.subsystems.gripper.GripperConstants;
+import frc.robot.subsystems.intake.Intake;
+import frc.robot.subsystems.shooter.Shooter;
+import frc.robot.subsystems.shooter.ShooterConstants;
+import java.util.function.Supplier;
+
+public class CommandGroups {
+    private static CommandGroups INSTANCE;
+    private final Intake intake;
+    private final Gripper gripper;
+    private final Elevator elevator;
+    private final Shooter shooter;
+    private final Conveyor conveyor;
+    private final GripperState gripperState;
+
+    private CommandGroups() {
+        intake = Intake.getInstance();
+        gripper = Gripper.getInstance();
+        elevator = Elevator.getInstance();
+        shooter = Shooter.getInstance();
+        conveyor = Conveyor.getInstance();
+        gripperState = new GripperState();
+    }
+
+    public static CommandGroups getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new CommandGroups();
+        }
+        return INSTANCE;
+    }
+
+    public Command elevatorGripperMinPosition() {
+        return Commands.parallel(
+                elevator.setHeight(ElevatorConstants.MIN_HEIGHT),
+                gripper.setWristPosition(GripperConstants.WRIST_FOLD_POSITION.mutableCopy()));
+    }
+
+    public Command intake() {
+        return elevatorGripperMinPosition()
+                .andThen(
+                        Commands.parallel(intake.intake(), gripper.intake())
+                                .until(gripper::hasNote)
+                                .andThen(intake.stop(), gripper.setRollerPower(0)));
+    }
+
+    public Command trapScore() {
+        return Commands.sequence(
+                gripper.setWristPosition(GripperConstants.WRIST_TRAP_ANGLE.mutableCopy()),
+                gripper.setRollerPower(GripperConstants.TRAP_POWER));
+    }
+
+    public Command gripperOuttake() {
+        return elevator.setHeight(ElevatorConstants.OUTTAKE_HEIGHT)
+                .onlyIf(gripperState::isGripperInsideRobot)
+                .andThen(gripper.outtake());
+    }
+
+    public Command shooterOuttake() {
+        return elevatorGripperMinPosition()
+                .andThen(
+                        Commands.parallel(
+                                gripper.setRollerPower(GripperConstants.OUTTAKE_POWER),
+                                conveyor.feed(),
+                                shooter.setVelocity(
+                                        () -> ShooterConstants.OUTTAKE_POWER.mutableCopy())));
+    }
+
+    public Command scoreCommandInit(Supplier<ScoreState> currentState) {
+        return currentState.get().initializeCommand();
+    }
+
+    public Command scoreSubsystemInit(Supplier<ScoreState> currentState) {
+        return currentState.get().initializeSubsystem();
+    }
+
+    public Command scoreExecute(Supplier<ScoreState> currentState) {
+        return currentState.get().score();
+    }
+
+    public Command climb() {
+        return null;
```

Comment:
Missing? Also, better to return `Commands.none()` for now so the code won't crash if that's used accidently before implementation.

Reply:
Not needed anymore
b515a7e